wizard/src/utils/heuristics.ts

```diff
@@ -14,13 +14,167 @@
+
+  const propDetected = new Map<string, number>();
+
+  detectedCountries.forEach((details: DetectedDetails, index: number) => {
+    const prop = details.detectedTypeProperty.dcProperty.dcid;
+    propDetected.set(prop, index);
+  });
+
+  if (propDetected.has("isoCode")) {
+    return propDetected.get("isoCode");
+  } else if (propDetected.has("countryAlpha3Code")) {
+    return propDetected.get("countryAlpha3Code");
+  } else if (propDetected.has("countryNumericCode")) {
+    return propDetected.get("countryNumericCode");
+  } else if (propDetected.has("name")) {
+    return propDetected.get("name");
+  }
+  return null;
+}
 
 /**
- * Given a csv, returns the predicted mapping
+ * Process all columns and return the one which best represents the detected
+ * Place along with its details. If no Place is detected, the return value is
+ * null.
  *
- * @param csv array of csv rows where each row is an array of strings
+ * @param cols is a mapping from column indices to (sampled) column values.
+ *  The indices correspond to those in the columnOrder Array.
+ * @param columnOrder is an ordered list (Array) of Columns.
+ * @param pDetector a PlaceDetector object.
+ *
+ * @returns a MappingVal object or null if no Place is detected.
  */
-export function getPredictions(csv: CsvData): Mapping {
+function detectPlace(
+  cols: Map<number, Array<string>>,
+  columnOrder: Array<Column>,
+  pDetector: PlaceDetector
+): MappingVal {
+  // Currently, only countries can be detected as Places.
+  const detectedCountries = new Map<number, DetectedDetails>();
+
+  cols.forEach((colVals: Array<string>, colIndex: number) => {
+    const pD = pDetector.detect(columnOrder[colIndex].header, colVals);
+    if (pD != null && pD.confidence == ConfidenceLevel.High) {
+      // Check if the detected Place is a Country.
+      if (pD.detectedTypeProperty.dcType.dcid === "Country") {
+        detectedCountries.set(colIndex, pD);
+      }
+    }
+  });
+
+  if (detectedCountries.size > 0) {
+    // Get the index of the detected property according to a preference order.
+    const index = countryOrder(detectedCountries);
+    if (index != null) {
+      return {
+        type: MappingType.COLUMN, // Place detection is only possible for columns.
+        column: columnOrder[index],
+        placeProperty:
+          detectedCountries.get(index).detectedTypeProperty.dcProperty,
+        placeType: detectedCountries.get(index).detectedTypeProperty.dcType,
+      };
+    }
+  }
   return null;
 }
+
+function detectDate(
+  cols: Map<number, Array<string>>,
+  columnOrder: Array<Column>,
+  dDetector: DateDetector
+): MappingVal {
+  const detectedDateColumns = new Array<Column>();
+  const detectedDateHeaders = new Array<Column>();
+
+  cols.forEach((colVals: Array<string>, colIndex: number) => {
+    const col = columnOrder[colIndex];
+
+    // Check if the column header can be parsed as a valid date.
+    if (dDetector.detectColumnHeaderDate(col.header) === true) {
```

Comment:
nit: don't think `=== true` is needed here and in line 118

Reply:
Done